src/metabase/integrations/ldap.clj

```diff
@@ -86,10 +86,12 @@
                (recur (json/parse-string new-value))
 
                (map? new-value)
+               ;; keys of new-value could be string or instance of DN
+               ;; in either cases they will be converted to a string before storing
                (do (doseq [k (keys new-value)]
-                     (when-not (DN/isValidDN (name k))
+                     (when-not (DN/isValidDN (if (keyword? k) (name k) (str k)))
```

Comment:
if the keys can be strings or DN instances, why are you checking for `keyword?` here?

maybe it would be simpler to first normalize the keys in the map to strings, and then run the existing logic as-is?

Reply:
Actually I just realized we don't need this anymore, I’ll revert it to the original code